lisa/tools/ethtool.py

```diff
@@ -0,0 +1,181 @@
+import re
+from pathlib import Path
+from typing import Any, List, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.util import LisaException
+
+
+class DeviceChannel:
+    # ethtool device channel info is in format -
+    # ~$ ethtool -l eth0
+    #   Channel parameters for eth0:
+    #   Pre-set maximums:
+    #   RX:             0
+    #   TX:             0
+    #   Other:          0
+    #   Combined:       1
+    #   Current hardware settings:
+    #   RX:             0
+    #   TX:             0
+    #   Other:          0
+    #   Combined:       1
+    __current_settings_pattern = re.compile(
+        r"Current hardware settings:(\s+)(?P<settings>.*?)$", re.DOTALL
+    )
+
+    __max_settings_pattern = re.compile(
+        r"Pre-set maximums:(\s+)(?P<settings>.*?)Current hardware settings:",
+        re.DOTALL,
+    )
+
+    __channel_count_param_pattern = re.compile(
+        r"(?P<param>Combined):( +)(?P<value>.*)$", re.MULTILINE
+    )
+
+    def __init__(self, device: str, device_channel_raw: str) -> None:
+        self.parse_channel_info(device, device_channel_raw)
+
+    def parse_channel_info(self, device: str, raw_str: str) -> None:
+        current_settings = self.__current_settings_pattern.match(raw_str)
+        max_settings = self.__max_settings_pattern.match(raw_str)
+        if (not current_settings) or (not max_settings):
+            raise LisaException(
+                "cannot get device channel current and/or max settings information"
+            )
+
+        current_param = self.__channel_count_param_pattern.match(
+            current_settings.group("settings")
```

Comment:
If one of current_settings or max_settings doesn't exists, what happens to call `.group`?